pypose/module/dynamics.py

```diff
@@ -6,7 +6,587 @@
+
+    Note:
+        In this general example, all variables are in a batch. User definable as appropriate.
+
+    Note:
+        More practical examples can be found at `examples/module/dynamics
+        <https://github.com/pypose/pypose/tree/main/examples/module/dynamics>`_.
+    '''
+    
+    def __init__(self, A, B, C, D, c1=None, c2=None):
+        super().__init__()
+        self.register_buffer('_A', A)
+        self.register_buffer('_B', B)
+        self.register_buffer('_C', C)
+        self.register_buffer('_D', D)
+        self.register_buffer('_c1', c1)
+        self.register_buffer('_c2', c2)
+
+    def forward(self, state, input):
+        r'''
+        Perform one step advance for the LTI system.
+
+        Args:
+            state (:obj:`Tensor`): The state (:math:`\mathbf{x}`) of the system 
+            input (:obj:`Tensor`): The input (:math:`\mathbf{u}`) of the system.
+
+        Returns:
+            ``tuple`` of Tensor: The state and observation of the system in next time step.
+        '''
+        return super().forward(state, input)
+
+    def state_transition(self, state, input):
+        r'''
+        Perform one step of LTI state transition.
+
+        .. math::
+            \mathbf{x}_{k+1} = \mathbf{A}\mathbf{x}_k + \mathbf{B}\mathbf{u}_k + \mathbf{c}_1
+
+        Args:
+            state (:obj:`Tensor`): The state (:math:`\mathbf{x}`) of the system 
+            input (:obj:`Tensor`): The input (:math:`\mathbf{u}`) of the system.
+
+        Returns:
+            ``Tensor``: The state the system in next time step.
+        '''
+        z = pp.bmv(self.A, state) + pp.bmv(self.B, input)
+        return z if self.c1 is None else z + self.c1
+
+    def observation(self, state, input):
+        r'''
+        Return the observation of LTI system.
+
+        .. math::
+            \mathbf{y}_k = \mathbf{C}\mathbf{x}_k + \mathbf{D}\mathbf{u}_k + \mathbf{c}_2
+
+        Args:
+            state (:obj:`Tensor`): The state (:math:`\mathbf{x}`) of the system 
+            input (:obj:`Tensor`): The input (:math:`\mathbf{u}`) of the system.
+
+        Returns:
+            ``Tensor``: The observation of the system in next time step.
+        '''
+        y = pp.bmv(self.C, state) + pp.bmv(self.D, input)
+        return y if self.c2 is None else y + self.c2
+
+    @property
+    def A(self):
+        r'''System transision matrix.'''
+        return self._A
+
+    @property
+    def B(self):
+        r'''System input matrix.'''
+        return self._B
+
+    @property
+    def C(self):
+        r'''System output matrix.'''
+        return self._C
+ 
+    @property
+    def D(self):
+        r'''System observation matrix.'''
+        return self._D
+
+    @property
+    def c1(self):
+        r'''Constant term generated by state-transition.'''
+        return self._c1
+
+    @property
+    def c2(self):
+        r'''Constant term generated by observation.
+        '''
+        return self._c2
+
+
+class LTV(LTI):
```

Comment:
@wang-chen I'm not sure whether I like this idea. My first glance is oh, we provide a LTV system for the user. Then, after looking into this class, I found not much difference with our previous base class of system. People sill need to inherit from this class to create their own. So I kind of think there is no necessary to do this at all. (This just add an addition layer in the middle without actually do much.). Also, the A, B, C, D properties interface seem to be missing, which should be provided.